service/start.go

```diff
@@ -13,68 +14,67 @@ import (
 	"github.com/spf13/viper"
 )
 
-// Start a service
-func (service *Service) Start() (serviceIDs []string, err error) {
+// Start starts the service.
+func (service *Service) Start() ([]string, error) {
 	status, err := service.Status()
 	if err != nil || status == RUNNING {
-		return //TODO: if the service is already running, serviceIDs should be returned.
+		return nil, err //TODO: if the service is already running, serviceIDs should be returned.
 	}
 	// If there is one but not all services running stop to restart all
 	if status == PARTIAL {
-		err = service.StopDependencies()
-		if err != nil {
-			return
+		if err = service.StopDependencies(); err != nil {
+			return nil, err
 		}
 	}
 	networkID, err := container.CreateNetwork(service.namespace())
 	if err != nil {
-		return
+		return nil, err
 	}
-	dependenciesFromService := service.DependenciesFromService()
-	serviceIDs = make([]string, len(dependenciesFromService))
-	var mutex sync.Mutex
-	var wg sync.WaitGroup
+	var (
+		mutex                   sync.Mutex
+		wg                      sync.WaitGroup
+		dependenciesFromService = service.DependenciesFromService()
+		serviceIDs              = make([]string, len(dependenciesFromService))
+	)
 	for i, dependency := range dependenciesFromService {
 		wg.Add(1)
 		go func(dep *DependencyFromService, i int) {
-			defer wg.Done()
 			serviceID, errStart := dep.Start(networkID)
 			mutex.Lock()
-			defer mutex.Unlock()
 			serviceIDs[i] = serviceID
 			if errStart != nil && err == nil {
 				err = errStart
 			}
+			mutex.Unlock()
```

Comment:
I added a comment on the original PR. Is there a reason not to use the defer ?
Except there is a good reason I would like to keep the unlock and done as soon as possible because we might have some deadlocks if one day we add some returns in between and forget about that

Reply:
My motivation was to get rid of defer call in simple function with zero/one return statement and I find it more readable, feel free to change it.

Reply:
I'm putting back the defers